pkg/storage/unified/resource/search.go

```diff
@@ -568,7 +566,12 @@ func (s *searchSupport) build(ctx context.Context, nsr NamespacedResource, size
 				}
 
 				// Update the key name
-				key.Name = iter.Name()
```

Comment:
BuildDocument takes a pointer the the key, meaning that key.Name kept on updating the same key.